### ISSUE TITLE
Got X96 max+ (2+16G) working with LAN+WIFI but still some confused points

### ISSUE THREAD

**Issue by dinhchinh82** [ophub/amlogic-s9xxx-armbian#95]
@ophub I've tried your latest build for S905X2 (2G+16G, Realtek card), and got both ethernet and wifi working from USB after using the dtb file from balbes150 (5.7.8)
Armbian_22.02.0_Aml_s905x3_focal_5.4.175_2022.01.30.0137.img.gz

Here is the detail log:
http://ix.io/3NXY

Here are updates board photos: 
![IMG_5628](https://user-images.githubusercontent.com/22479095/151687846-036f752d-6848-4752-b73c-5ffb6c5ed247.JPG)

![IMG_5631](https://user-images.githubusercontent.com/22479095/151687842-06a83bee-00fb-41d1-a132-9abe7096a778.JPG)

But the log related to the network adapter is showed continuously. I'm not sure how to disable this log.
![IMG_5630](https://user-images.githubusercontent.com/22479095/151687847-62a7693b-15d5-41e4-980d-84b785855a90.JPG)

By the way, I just confused that you said Realtek has no driver under Armbian but it seems working now with balbes150's dtb.
https://github.com/ophub/amlogic-s9xxx-armbian/issues/47#issuecomment-987565762
S905x3 dtb file from balbes150's image:
https://www.dropbox.com/s/d83yqv69tkjvias/meson-sm1-sei610_578.dtb?dl=0

Could you give me any suggestions for this board?

Thank you.

**Comment by dinhchinh82** [comment]
Unfortunately, after install armbian to eMMC and unplug the USB and reboot again, my S905X3 box is unable to boot:
here is the screen log:
armbian-install

Note:
before running install armbian to eMMC, I already replaced your original S905x3's dtb for X96-Max+  (meson-sm1-x96-max-plus.dtb) by balbes150's dtb meson-sm1-sei610_578.dtb as mentioned above.

**Comment by ophub** [comment]
flippy has added the `rtl8188fu/8822cs/8189fs` driver to the kernel, which is enabled by default in `5.10.91/5.4.172/5.15.14 and later kernels`.

You can test the wireless network with the latest kernel.

If you use another `dtb` yourself, modify the dtb in the installation file `armbian-install`, so that it can be correct when writing emmc.

**Comment by dinhchinh82** [comment]
@ophub thank you for your quick comment.

>Choose the correct firmware. What you choose is the dtb of s905x3, if your box is s905x2, use your own series. I think the solid one comes with it and it can be used.

I can make sure my box is X96 max + (2G + 16G) with S905x3 CPU.

> flippy has added the rtl8188fu/8822cs/8189fs driver to the kernel, which is enabled by default in 5.10.91/5.4.172/5.15.14 and later kernels.
Yes, I found out the wifi works well with default dtb while running from USB.
By the way, there is very strange that after install to eMMC, the box is unable to boot and got the the error below EVEN with default dtb of X96 max + (index 51).

![IMG_5632](https://user-images.githubusercontent.com/22479095/151689182-2ee49a31-2618-4ac5-9354-da43e81ba368.jpg)

I also tried to install to eMMC using X96 max + overclock (index 52) but also got the same issue.

In summary, I just confused why the box running on USB but unable to boot after installing to eMMC.

Is there a way to burn exactly default setting of USB to eMMC without any selection of board?

Thank you.

**Comment by ophub** [comment]
https://github.com/ophub/amlogic-s9xxx-armbian/blob/3a9aac5b59ea534a7b468bce7bddca5fe31a9800/build-armbian/common-files/files/usr/sbin/armbian-install#L102

```
armbian-install
select 51
```

**Comment by dinhchinh82** [comment]
> armbian-install
> select 51

My box is S905X3 and I already selected 51 without any changes of DTB (don't use DTB from balbes150 anymore) but still got error after booting from eMMC.

I have several this box (S905X3 with 2+16G) and I can make sure there is no S905X2.

**Comment by ophub** [comment]
Yours is a 100M network？

**Comment by dinhchinh82** [comment]
my LAN is 1Gbs network

**Comment by dinhchinh82** [comment]
Please check my updated photos:
![IMG_5640](https://user-images.githubusercontent.com/22479095/151697030-4f381218-e865-4487-9e6e-cd64edb5837d.JPG)
![IMG_5639](https://user-images.githubusercontent.com/22479095/151697037-93698b9b-2f5a-4f00-aec8-332c79159082.JPG)

**Comment by dinhchinh82** [comment]
There are several version of X96 max plus (S905X3), this guy also have same version as mine:
https://forum.freaktab.com/forum/tv-player-support/amlogic-based-tv-players/s905x3/firmware-roms-tools-ds/805518-x96-max-plus-2gb-optical-rt-8189ftv-q5x3_141-v3-1

https://androidtvbox.eu/x96-max-plus2-firmware-for-tv-box-with-s905x3-soc-20200901/
I can restore my box with the Android version from link above.

**Comment by ophub** [comment]
https://github.com/ophub/amlogic-s9xxx-armbian/blob/3a9aac5b59ea534a7b468bce7bddca5fe31a9800/build-armbian/common-files/files/usr/sbin/armbian-install#L102

`armbian-install`: Change the dtb in `line 102` to `meson-sm1-x96-max-plus-100m.dtb`, and then install and write emmc, which is the dtb file used when usb starts.

**Comment by dinhchinh82** [comment]
>armbian-install: Change the dtb in line 102 to meson-sm1-x96-max-plus-100m.dtb, and then install and write emmc, which is the dtb file used when usb starts.

I'm trying again.

By the way, I just confused why do you need to add an resistor as your photo?
https://user-images.githubusercontent.com/68696949/110910162-ec967000-834b-11eb-8fa6-64727ccbe4af.jpg

As before, when I used balbes150's armbian image, sometimes my X96 max plus+ unable to reboot after power outtage.

**Comment by ophub** [comment]
When the 5.10 kernel just came out, because u-boot was easily disturbed when it was booted, adding a resistor could solve this problem. At present this has been automatically resolved by u-boot reload, but this resistor no longer needs to be added.

**Comment by dinhchinh82** [comment]
> When the 5.10 kernel just came out, because u-boot was easily disturbed when it was booted, adding a resistor could solve this problem. At present this has been automatically resolved by u-boot reload, but this resistor no longer needs to be added.

So if using the kernel 5.10.91/5.4.172/5.15.14 or later, there is no need to add this resistor anymore, right?

**Comment by ophub** [comment]
No need, it's been many generations since 5.10, now it's 5.16, this was a moment, about the summer of 2020.

**Comment by dinhchinh82** [comment]
@ophub  I've tried the following steps and the box still could not boot from eMMC after running armbian-install:

Step 1: Burning the S905x3 images to USB
Armbian_22.02.0_Aml_s905x3_bullseye_5.15.16_2022.01.22.1251.img.gz

Step 2: Boot from USB with default dtb, the LAN is not recognized

Step 3: Edit dtb from uEnv.txt
from meson-sm1-x96-max-plus-100m.dtb to meson-sm1-sei610.dtb
then reboot

Step 4: after rebooting, the LAN is recognized 

<img width="515" alt="Screen Shot 2022-01-30 at 23 31 50" src="https://user-images.githubusercontent.com/22479095/151708510-3d753eff-3bbc-4f0c-b7cc-c10e5a513869.png">

Step 5: Edit /usr/sbin/armbian-install file as below:
<img width="960" alt="Screen Shot 2022-01-30 at 23 33 00" src="https://user-images.githubusercontent.com/22479095/151708540-ea4d91db-d473-4949-beee-182a45446833.png">

Step 6: Run the armbian-install
<img width="651" alt="Screen Shot 2022-01-30 at 23 32 28" src="https://user-images.githubusercontent.com/22479095/151708580-0cb78f26-1ba6-41d2-9d5d-cb6f6918a0e7.png">

<img width="741" alt="Screen Shot 2022-01-30 at 23 32 37" src="https://user-images.githubusercontent.com/22479095/151708592-19a2e3e1-afcf-4b7f-ab98-1aa0a6134dc2.png">

Step 7: Run "shutdown now" command, unplug power, unplug USB

Step 8: Replug power. There is no any boot/logo or terminal on screen.

So I still confused while the box running well on USB but not work on eMMC.

Please give me your suggestion if any.

Thank you so much.

**Comment by dinhchinh82** [comment]
`51:X96 Max+:s905x3:meson-sm1-x96-max-plus.dtb:u-boot-x96maxplus.bin:/lib/u-boot/x96maxplus-u-boot.bin.sd.bin:/lib/u-boot/hk1box-bootloader.img:4C@2100Mhz,4GB Mem,1Gb Net `

>armbian-install: Change the dtb in line 102 to meson-sm1-x96-max-plus-100m.dtb, and then install and write emmc, which is the dtb file used when usb starts.

I also tried your suggestion above to use meson-sm1-x96-max-plus-100m.dtb but the box is also still unable to boot (black screen) after unplug USB and reboot:
<img width="668" alt="Screen Shot 2022-01-30 at 23 57 41" src="https://user-images.githubusercontent.com/22479095/151709462-8cefae3b-c4b1-46d6-83fa-ff6ea362d050.png">
<img width="743" alt="Screen Shot 2022-01-31 at 00 02 07" src="https://user-images.githubusercontent.com/22479095/151709467-0f587d54-6d0b-4ca7-a9cc-a0dd80fe65ac.png">

**Comment by ophub** [comment]
You use the firmware of the `5.4 kernel` and modify it to use your dtb. do not use other kernels

**Comment by dinhchinh82** [comment]
@ophub I will try again.
By the way, how to boot from SD card?

**Comment by dinhchinh82** [comment]
@ophub I already tried to use 5.4 kernel but still got black screen error after install to eMMC:
<img width="608" alt="Screen Shot 2022-01-31 at 08 25 53" src="https://user-images.githubusercontent.com/22479095/151727111-8a8bbc1c-07b4-4554-8595-83b9f9ff424d.png">

I also added your suggestion to the script:
`59:X96 Max+:s905x3:meson-sm1-sei610.dtb:u-boot-x96maxplus.bin:NA:NA:4C@2100Mhz,2g/16g,100m
`
<img width="619" alt="Screen Shot 2022-01-31 at 08 20 40" src="https://user-images.githubusercontent.com/22479095/151727122-ac2f167f-b19b-4319-b1e1-c89c43f1b5a3.png">

The strange error message still happend:

> Please Input ID: 59
> FDTFILE: [ meson-sm1-sei610.dtb ]
> UBOOT_OVERLOAD: [ u-boot-x96maxplus.bin ]
> Start backup default bootloader
> 4+0 records in
> 4+0 records out
> 4194304 bytes (4.2 MB, 4.0 MiB) copied, 0.24677 s, 17.0 MB/s
> Error: /dev/mmcblk2: unrecognised disk label

![Uploading Screen Shot 2022-01-31 at 08.24.48.png…]()

**Comment by ophub** [comment]
Can I use the 5.4 kernel to write emmc?

**Comment by dinhchinh82** [comment]
> Can I use the 5.4 kernel to write emmc?

Yes. I can use 5.4 kernel from USB but after install to eMMC, the box is still unable to boot.

Here is the list of partitions before running armbian-install:
<img width="443" alt="Screen Shot 2022-01-31 at 08 56 40" src="https://user-images.githubusercontent.com/22479095/151729124-9daa3f6c-061a-483d-a440-36e0f2d33279.png">

**Comment by dinhchinh82** [comment]
After install to eMMC, it is unable to boot from USB again to check the data from eMMC. It's quite strange again.

As before, when I use images from balbe150,  it is able to boot from SD card even after installing to eMMC.

**Comment by dinhchinh82** [comment]
This is updated screenshot when I tried to add some debug code to your armbian-install to list the latest partitions
<img width="740" alt="Screen Shot 2022-01-31 at 09 51 12" src="https://user-images.githubusercontent.com/22479095/151732669-2bc3ffdb-65f8-445c-b64a-f5ba37fdda42.png">

**Comment by ophub** [comment]
After the installation is completed using armbian-install, you must `unplug` the usb/tf/sd card, reinsert the power supply and boot from emmc

**Comment by dinhchinh82** [comment]
> After the installation is completed using armbian-install, you must `unplug` the usb/tf/sd card, reinsert the power supply and boot from emmc

Sure, I always unplug the usb/tf/sd card then reinsert the power supply but still no signal.

**Comment by ophub** [comment]
Is it so magical that even 5.4 cannot be written to emmc? that can only be used in usb

**Comment by dinhchinh82** [comment]
@ophub I just tried to installed your build (5.10.93) to X96 max+ (4G+32G 1Gps) using index 51 and it works well from eMMC.

Only X96 max + version 2+16 (100Mps) got issue that unable to boot from eMMC. I also very confused why it happens.

Please note that my X96 max+ (2+16, 100Mps) still can boot from eMMC if using Balbes150 image.

**Comment by dinhchinh82** [comment]
`59:X96 Max+:s905x3:meson-sm1-sei610.dtb:u-boot-x96maxplus.bin:NA:NA:4C@2100Mhz,2g/16g,100m`

@ophub Can you help me to make the profile that exactly same as the profile for default USB image of X96 max +?

**Comment by ophub** [comment]
Whether it can be started is determined by two files, one is the correct dtb and the other is the correct u-boot. Because the 5.4 kernel only uses dtb, it is recommended that you test with 5.4 to determine which dtb is suitable for your box.

Which dtb and u-boot.ext file do you use in the Balbes150 firmware, you can copy and test.

**Comment by dinhchinh82** [comment]
@ophub so your u-boot is .bin file, right?

Is there a way to boot your image from micro SD card? 

I used to boot Balbes150's image from micro SD card on X96 max +.

**Comment by ophub** [comment]
.bin file in the /boot directory

In the available boxes, tf/sd/usb/emmc boot is supported

**Comment by dinhchinh82** [comment]
> In the available boxes, tf/sd/usb/emmc boot is supported

There is another strange point that I've tried to boot your firmware from TF card but unable to boot, only could boot from USB 3.0 port.
But the firmware from Balbes150 could boot from TF card. Please note that I always restore default Android firmware before testing any Armbian firmware.

**Comment by ophub** [comment]
It can be used from tf card, my s905x3 x96max+ box is used on tf card.

**Comment by dinhchinh82** [comment]
@ophub I'd like to give you the summary of the current test for S905X3 X96 max+ (2G+16G + 100Mbp LAN)

1. Armbian_22.02.0_Aml_s905x3_bullseye_5.4.176: default dtb (meson-sm1-x96-max-plus-100m.dtb)
- Boot from TF card: unable to boot
- Boot from USB 3.0 port: can boot and login to Armbian, LAN did not work, WIFI works
- eMMC: unable to boot
2. Armbian_22.02.0_Aml_s905x3_bullseye_5.4.176: custom dtb (meson-sm1-sei610.dtb)
- Boot from TF card: unable to boot
- Boot from USB 3.0 port: can boot and login to Armbian, LAN works, WIFI works
- eMMC: unable to boot
3. Armbian_22.02.0_Aml_s905x_bullseye_5.10.96: default dtb (meson-sm1-x96-max-plus-100m.dtb)
- Boot from TF card: unable to boot
- Boot from USB 3.0 port: can boot and login to Armbian, LAN did not work, WIFI works
- eMMC: unable to boot
4. Armbian_22.02.0_Aml_s905x_bullseye_5.10.96: custom dtb (meson-sm1-sei610.dtb)
- Boot from TF card: unable to boot
- Boot from USB 3.0 port: can boot and login to Armbian, LAN works, WIFI works
- eMMC: unable to boot
5. Armbian_20.10_Arm-64_bullseye_current_5.9.0.img.xz:  default dtb (mmeson-sm1-sei610.dtb
- Boot from TF card: can boot and login to Armbian, LAN work (but sometimes not stable), WIFI works
- Boot from USB 3.0 port: unable to boot
- eMMC: can boot and login to Armbian, LAN work (but sometimes not stable), WIFI works

Please give me any other suggestions to test if any. I'd like to install the Armbian to eMMC for long time runing.
There is no problem for me to install any additional tools for debugging according to your requirement. I have some knowledge in Linux, Android, Flashing firmware etc.
Thank you.

**Comment by ophub** [comment]
dtb: /boot/uEnv.ext   meson-sm1-sei610.dtb
uboot: /boot/u-boot-s905x2-s922.bin copy to u-boot.ext

**Comment by dinhchinh82** [comment]
@ophub 

> dtb: /boot/uEnv.ext meson-sm1-sei610.dtb uboot: /boot/u-boot-s905x2-s922.bin copy to u-boot.ext

The /boot/u-boot-s905x2-s922.bin is the old one from Balbes150 or new one from Flippy? I'm afraid that if using the old one the issue related to LAN will be occurred again even the box could boot.
